product_internal_reference_code/models/product_product.py

```diff
@@ -11,9 +11,24 @@ class ProductProduct(models.Model):
     @api.model
     def create(self, vals):
         if not vals.get('default_code', False):
-            internal_code_prefix = \
-                self.env.user.company_id.internal_code_prefix or ''
-            seq_code = self.env['ir.sequence'].next_by_code(
+            # Get the sequence with user company_id
+            seq_code = self.env['ir.sequence'].with_context(
+                force_company=self.env.user.company_id.id).next_by_code(
                 'product.product.internal_code')
-            vals.update({'default_code': internal_code_prefix + seq_code})
+            # If there is not seq_code, create a new one for the company
+            if not seq_code:
+                seq = {
+                    'name': self.env.user.company_id.name + ' - Product Sequence',
+                    'code': 'product.product.internal_code',
+                    'prefix': self.env.user.company_id.internal_code_prefix
+                              or '',
+                    'padding': 5,
+                    'number_increment': 1,
+                    'company_id': self.env.user.company_id.id
+                }
+                self.env['ir.sequence'].create(seq)
+                seq_code = self.env['ir.sequence'].with_context(
+                    force_company=self.env.user.company_id.id).next_by_code(
+                    'product.product.internal_code')
```

Comment:
In what case there is no seq_code if ir.sequence record is generated when company is created?

Reply:
The first company?

Reply:
In that case, isn't creating an ir.sequence record for each company at installation of the module more appropriate?

Reply:
You mean having a post_hook method to create the ir.sequence?

Reply:
Yes